# Import libaries

In [2]:
import kagglehub
import pandas as pd

from pathlib import Path

# 0. Load data

In [4]:
path = kagglehub.dataset_download(
    "jainilcoder/netflix-stock-price-prediction",
    output_dir="data"
)

DATASET_DIR = Path("data")
DATA_DIR = DATASET_DIR / "NFLX.csv"

print("Path to dataset files:", DATA_DIR)

Path to dataset files: data\NFLX.csv


# 1. Data inspection

In [21]:
df = pd.read_csv(DATA_DIR)

print("Shape:", df.shape)
print("Columns:", list(df.columns))
df.head(1)

Shape: (1009, 7)
Columns: ['Date', 'Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume']


,Date,Open,High,Low,Close,Adj Close,Volume
0,2018-02-05,262.0,267.899994,250.029999,254.259995,254.259995,11896100


### Info
The dataset contains **1,009** records of Netflix stock data, including `Open`, `Close`, `High`, `Low`, `Volume`, and `Date` features. The dataset is relatively small, but it is suitable for this project because it is my first Time Series project. It provides a good opportunity to learn how LSTM and GRU models work with sequential data.

In [22]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1009 entries, 0 to 1008
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       1009 non-null   str    
 1   Open       1009 non-null   float64
 2   High       1009 non-null   float64
 3   Low        1009 non-null   float64
 4   Close      1009 non-null   float64
 5   Adj Close  1009 non-null   float64
 6   Volume     1009 non-null   int64  
dtypes: float64(5), int64(1), str(1)
memory usage: 55.3 KB


### Info
There are no missing values in the dataset. It contains 6 numerical features and 1 string feature, which is the `Date` column.

In [23]:
df.describe()

,Open,High,Low,Close,Adj Close,Volume
count,1009.000000,1009.000000,1009.000000,1009.000000,1009.000000,1.009000e+03
mean,419.059673,425.320703,412.374044,419.000733,419.000733,7.570685e+06
std,108.537532,109.262960,107.555867,108.289999,108.289999,5.465535e+06
min,233.919998,250.649994,231.229996,233.880005,233.880005,1.144000e+06
25%,331.489990,336.299988,326.000000,331.619995,331.619995,4.091900e+06
50%,377.769989,383.010010,370.880005,378.670013,378.670013,5.934500e+06
75%,509.130005,515.630005,502.529999,509.079987,509.079987,9.322400e+06
max,692.349976,700.989990,686.090027,691.690002,691.690002,5.890430e+07


### Info
The features have a wide range of values, which is expected because Netflix's stock price has experienced many ups and downs over the years. The data will need to be scaled before training the model to make the training process more stable and efficient.